pytest_fixtures/component/puppet.py

```diff
@@ -114,3 +114,14 @@ def module_puppet_classes(
             f'and environment = {module_env_search.name}'
         }
     )
+
+
+@pytest.fixture(scope="module")
+def module_default_puppet_proxy(module_puppet_enabled_sat):
+    """Use the default installation puppet smart proxy"""
+    proxy = (
+        module_puppet_enabled_sat.api.SmartProxy()
+        .search(query={'search': f'url = {module_puppet_enabled_sat.url}:9090'})[0]
+        .read()
+    )
+    return module_puppet_enabled_sat.api.SmartProxy(id=proxy.id).read()
```

Comment:
You dont need double read here you have done reading in previous step!

Reply:
already fixed in actual fixture